prov/rxm/src/rxm_init.c

```diff
@@ -46,7 +46,9 @@ int rxm_info_to_core(uint32_t version, const struct fi_info *hints,
 		     struct fi_info *core_info)
 {
 	/* Support modes that ofi_rxm could handle */
-	if (FI_VERSION_GE(version, FI_VERSION(1, 5)))
+	if (FI_VERSION_GE(version, FI_VERSION(1, 5)) &&
+	    (!hints || !hints->domain_attr ||
+	     !(hints->domain_attr->mr_mode & (FI_MR_BASIC | FI_MR_SCALABLE))))
 		core_info->domain_attr->mr_mode |= FI_MR_LOCAL;
 	else
 		core_info->mode |= FI_LOCAL_MR;
```

Comment:
Should we remove other Basic MR flags (`FI_MR_VIRT_ADDR`, `FI_MR_ALLOCATED`, and `FI_MR_PROV_KEY`) in case of `mr_mode & FI_MR_BASIC` provided? Since setting of these flags are equivalent of setting of `FI_MR_BASIC` flag

Reply:
If the app is providing FI_MR_BASIC it should not specify any other MR mode bits. So, removing them may not be needed.

Reply:
Ok. Maybe the right way is to not add `FI_MR_PROV_KEY` during opening core provider domain